src/coreclr/jit/lowerxarch.cpp

```diff
@@ -1180,6 +1180,13 @@ GenTree* Lowering::LowerHWIntrinsic(GenTreeHWIntrinsic* node)
 
             if (op1IsVectorZero && op2IsVectorZero)
             {
+                // We need to change op1's type to the return type of the node in this case.
+                // We have to do this because we are going to propagate the constant up.
+                op1->ChangeType(node->TypeGet());
+
+                // Ensure the upper values are zero by zero-initialization.
+                op1->AsVecCon()->gtSimd32Val = {};
```

Comment:
Maybe there should be a `GenTreeVecCon::SetZero()` function to encapsulate this, adjacent to the existing `IsZero()`?

Reply:
Probably would be a good idea to do.